test/e2e/specs/editor/various/block-deletion.spec.js

```diff
@@ -0,0 +1,350 @@
+		// Ensure the caret is in a correct position.
+		await page.keyboard.type( '| ← caret was here' );
+		await expect(
+			editor.canvas
+				.getByRole( 'document', {
+					name: 'Paragraph block',
+				} )
+				.last()
+		).toContainText( 'Second| ← caret was here' );
+	} );
+
+	test( 'deleting last selected block via backspace', async ( {
+		editor,
+		page,
+	} ) => {
+		// Add a couple of paragraphs.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'First' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Second' },
+		} );
+
+		// Add the Image block and focus it.
+		await editor.insertBlock( {
+			name: 'core/image',
+		} );
+		const imageBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Image',
+		} );
+		await expect(
+			imageBlock.getByRole( 'button', { name: 'Upload' } )
+		).toBeFocused();
+		await page.keyboard.press( 'ArrowUp' );
+		await expect( imageBlock ).toBeFocused();
+
+		// Hit backspace and ensure the Image block was removed.
+		await page.keyboard.press( 'Backspace' );
+		await expect(
+			editor.canvas
+				.getByRole( 'document', {
+					name: 'Paragraph block',
+				} )
+				.last()
+		).toContainText( 'Second' );
+
+		// Ensure the caret is in a correct position.
+		await page.keyboard.type( '| ← caret was here' );
+		await expect(
+			editor.canvas
+				.getByRole( 'document', {
+					name: 'Paragraph block',
+				} )
+				.last()
+		).toContainText( 'Second| ← caret was here' );
+	} );
+
+	test( 'deleting last two selected blocks via backspace', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Add a couple of paragraphs.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'First' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Second' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Third' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+
+		// Ensure the empty paragraph is focused.
+		await expect(
+			editor.canvas.getByRole( 'document', {
+				name: /Empty block/i,
+			} )
+		).toBeFocused();
+
+		// Select the last two paragraphs.
+		await pageUtils.pressKeyWithModifier( 'shift', 'ArrowUp' );
+		await expect(
+			editor.canvas.locator( 'p.is-multi-selected' )
+		).toHaveCount( 2 );
+
+		// Hit backspace and ensure the last two paragraphs were deleted.
+		await page.keyboard.press( 'Backspace' );
+		await expect(
+			editor.canvas
+				.getByRole( 'document', {
+					name: 'Paragraph block',
+				} )
+				.last()
+		).toContainText( 'Second' );
+
+		// Ensure a new empty block was created and focused.
+		await expect.poll( editor.getBlocks ).toHaveLength( 3 );
+		await expect(
+			editor.canvas.getByRole( 'document', {
+				name: /Empty block/i,
+			} )
+		).toBeFocused();
+	} );
+
+	test( 'deleting all blocks', async ( { editor, page } ) => {
```

Comment:
Let's keep the original title here for clarity: `results in the default block getting selected`.

Reply:
Not sure about this one, as it wouldn't match the format of the rest of the titles: 

```
Block deletion › deleting the last block via its options menu
Block deletion › deleting the last block via the keyboard shortcut
Block deletion › deleting the last block via backspace from an empty paragraph
Block deletion › deleting last selected block via backspace
Block deletion › deleting the last two selected blocks via backspace
Block deletion › deleting all blocks
Block deletion › deleting all blocks when the default block is unavailable
```

I assumed that adding the `...results in...` to every title would ultimately be less readable. We did that in the old `block-deletion.test.js`, where each `test` was in a separate `describe` to (I guess) prevent the titles from reaching 300 cols 😅, e.g.:

```
block deletion - deleting the third and fourth blocks using backspace with multi-block selection results in two remaining blocks and positions the caret at the end of the second block
```

IMO, the expected result is described sufficiently with the `expect` calls. What do you think?

Reply:
I personally don't mind if the title is long or not 😅, but I see what you mean!